unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/BodyShapeController.cs

```diff
@@ -42,89 +51,79 @@ public override void Load(string bodyShapeId, Transform parent, Action<WearableC
 
     public void SetActiveParts(bool lowerBodyActive, bool upperBodyActive, bool feetActive)
     {
-        lowerBodyRenderer.gameObject.SetActive(lowerBodyActive);
-        lowerBodyRenderer.enabled = lowerBodyActive;
-
-        upperBodyRenderer.gameObject.SetActive(upperBodyActive);
-        upperBodyRenderer.enabled = upperBodyActive;
-
-        feetRenderer.gameObject.SetActive(feetActive);
-        feetRenderer.enabled = feetActive;
+        this.lowerBodyActive = lowerBodyActive;
+        this.upperBodyActive = upperBodyActive;
+        this.feetActive = feetActive;
     }
 
     public void SetupEyes(Material material, Texture texture, Texture mask, Color color)
     {
-        if (assetContainer?.transform == null)
+        if (assetContainer != null && assetContainer.transform == null)
         {
             Debug.LogWarning("Tried to setup eyes when the asset not ready");
             return;
         }
 
+        material.SetTexture(AvatarUtils._EyesTexture, texture);
+        material.SetTexture(AvatarUtils._IrisMask, mask);
+        material.SetColor(AvatarUtils._EyeTint, color);
```

Comment:
It turned out that setting the same material inside the recursive function was not needed... it's enough to set the material once.